Sender_test.cpp

```diff
@@ -0,0 +1,40 @@
+#include <string>
```

Comment:
I see a very good job in the Test segregation, its interesting how you cover different aspects inside any section of your test case